spec/support/ssl_helper.rb

```diff
@@ -12,16 +12,41 @@ def initialize
     def next
       @iter.next
     rescue StopIteration
-      @keystash << OpenSSL::PKey::RSA.new(2048)
+      @keystash << generate_key
       save
       @keystash.last
     end
 
+    def generate_key
+      case (rand * 4).to_i
+      when 0
+        OpenSSL::PKey::RSA.new(2048)
+      when 1
+        generate_ecdsa_key('prime256v1')
+      when 2
+        generate_ecdsa_key('secp384r1')
+      when 3
+        generate_ecdsa_key('secp521r1')
+      end
+    end
+
+    def generate_ecdsa_key(curve)
+      k = OpenSSL::PKey::EC.new(curve)
+      k.generate_key
+      Acme::Client::CertificateRequest::ECKeyPatch.new(k)
```

Comment:
Oh I see. It worked because its fixed in the latest ruby releases.

I tried again with `2.1` and I got the same issue as you.

Shouldn't the client wrap the `OpenSSL::PKey::EC` with `ECKeyPatch` whenever its needed then?

It just feel strange to me that we expose the developers to this. Especially knowing that this patch will not be required in future ruby.